crates/phactory/src/prpc_service.rs

```diff
@@ -1068,27 +1057,27 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         let challenge_handler = request.decode_challenge_handler().map_err(from_display)?;
         let block_sec = system.now_ms / 1000;
         let block_number = system.block_number;
-        let attestation = if dev_mode {
+        let attestation = if dev_mode || attestation_provider == "" || attestation_provider == "none" {
```

Comment:
changed as your suggest, but I use `serde_json` to do `enum` <-> `String`, any good idea?

Reply:
Looks like we can just define it in Protobuf.
https://developers.google.com/protocol-buffers/docs/proto3#enum

Reply:
AttestationProvider enum also using in pallets, I'm thinking defined a global type in protobuf isn't a good idea?

Reply:
I mean they can both defined in protobuf and the phala-types. convert them by hand when needed. It's better than serde_json because it's statically type-checked.
If we want to avoid the redundant defination, we can use our customized type decoration syntax: https://github.com/Phala-Network/prpc-protos/blob/master/pruntime_rpc.proto#L188. This allows us to reference Rust types in Rust.

Reply:
got it, I'll check https://github.com/Phala-Network/prpc-protos/blob/master/pruntime_rpc.proto#L188
BTW, `None` already in-use in proto, it seems I can't use it for my `AttestationProvider`, I still think `OptOut` is a good name, or do you have any suggest for that?

Reply:
The best choice in my mind is Option<AttestationProvider>

Reply:
Have you tried? :
```proto
  // Attestation provider;
  // @codec scale Option<phala_types::AttestationProvider>
  bytes attestation_provider = 7;
```

Reply:
I think I can try the second one, I don't think `Option` is good for the case

Reply:
> Option<AttestationProvider> made each use of AttestationProvider, need check None first, then match enum

Can be written as:
```
match provider {
  None => ...,
  Some(Sgx) => ...,
  Some(Dcap) => ...,
}

No too much difference.

Reply:
Refactor including Phala pallet with this style